src/components/MapToQueryGeometry/MapToQueryGeometry.vue

```diff
@@ -73,12 +74,14 @@
       map.on('idle', () => {
         const { layer, id } = this.mapboxQueryData
         if (!map.getLayer(layer)) { return }
-        const AREA_KEY = AREA_KEY_MAP[this.areaType]
-        const geometry = map.querySourceFeatures(layer, { sourceLayer: layer })
-          // Forcing a string here by doing `+ ''`
-          .find(({ properties }) => properties?.[AREA_KEY] + '' === id)
```

Comment:
This only returned one feature, where It can be multiple features. This was causing the partial regions showing up.